tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4246,6 +4247,90 @@ void CGMSHLSLRuntime::SetPatchConstantFunctionWithAttr(
   
 }
 
+// Returns true a global value is being updated
+static bool GlobalHasStoreUserRec(Value *V, std::set<Value *> &visited) {
+  bool isWriteEnabled = false;
+  if (V && visited.find(V) == visited.end()) {
+    visited.insert(V);
+    for (User *U : V->users()) {
+      if (isa<StoreInst>(U)) {
+        return true;
+      } else if (CallInst* CI = dyn_cast<CallInst>(U)) {
+        Function *F = CI->getCalledFunction();
+        if (!F->isIntrinsic()) {
+          HLOpcodeGroup hlGroup = GetHLOpcodeGroup(F);
+          switch (hlGroup) {
+          case HLOpcodeGroup::NotHL:
+            return true;
+          case HLOpcodeGroup::HLMatLoadStore:
+          {
+            HLMatLoadStoreOpcode opCode = static_cast<HLMatLoadStoreOpcode>(hlsl::GetHLOpcode(CI));
+            if (opCode == HLMatLoadStoreOpcode::ColMatStore || opCode == HLMatLoadStoreOpcode::RowMatStore)
+              return true;
+            break;
+          }
+          case HLOpcodeGroup::HLCast:
+          case HLOpcodeGroup::HLSubscript:
+            isWriteEnabled |= GlobalHasStoreUserRec(U, visited);

```

Comment:
Replace:
```
  isWriteEnabled |= GlobalHasStoreUserRec(U, visited);
```
with:
```
  if (GlobalHasStoreUserRec(U, visited))
    return true;
```
and return false at the end, then you can get rid of the local `isWriteEnabled` variable.